client/src/components/MainPage.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import TopNav from './TopNav';
+import CenterpieceDropDown from './CenterpieceDropDown';
+import axios from 'axios';
+const helper = require('./helper.js');
+
+const MainPage = (props) => {
+  const [locations, setLocations] = useState(['']);
+  const radianUnit = 2*Math.PI / locations.length;
+  const [userName, setUserName] = useState('');
+  const [currentLocation, setCurrentLocation] = useState(()=> location[0]);
+  const [friends, setFriends] = useState(['']);
+  const [suppress, setSuppress] = useState(false);
+
+  const createDimensions = (sideLength, padding) => {
+  const obj = {
+    componentSide: (sideLength + padding * 2),
+  }
+  return obj
+  }
+
+const dimensions = createDimensions(150, 300)
+
+const circle = {
+  centerX: dimensions.componentSide/2,
+  centerY: dimensions.componentSide/2,
+  radius: 350
+};
+
+const makeLocations = (locations) => {
+  const array = [];
+  for (let i = 0; i < locations.length; i++) {
+    let radian = radianUnit * i;
+    let point = helper.getPointsOnCircle([circle.centerX, circle.centerY], circle.radius, radian);
+    const location = (
+        <Text fill='antiquewhite' className='text' key={locations[i]} textAnchor='middle' x={point[0]} y={point[1]}>{locations[i]}</Text>
+    )
+    array.push(location);
+  }
+  return array;
+}
+
+const placeArm = (location) => {
+  let obj = {};
+  // loop thru locations
+  for (let i = 0; i < locations.length; i++) {
+    // if location equals locations[i]
+    if(location === locations[i]) {
+      // use i to calculate radian and call helper
+      let radian = radianUnit * i;
+      let point = helper.getPointsOnCircle([circle.centerX, circle.centerY], circle.radius, radian);
+      obj['x'] = point[0];
+      obj['y'] = point[1];
+    }
+  }
+  return obj;
+}
+
+useEffect( () => {
+    axios.get('/api/users')
+  .then((res) => {
+    let oneUser = res.data;
+    console.log('this is oneUser: ', oneUser)
+    setLocations(oneUser.locations);
+    setUserName(oneUser.userName);
+    setCurrentLocation(oneUser.currentLocation)
+    setFriends(oneUser.friends)
+  })
+  .catch(err => {throw err;})
+},[]);
+
+const Arm = () => {
+  const armLocation = placeArm(currentLocation);
```

Comment:
placeArm isn't really *placing* anything, it's just computing the coordinates for the arm. So maybe a better name would be `computeArmCoordinates` or something?